merkledag/node.go

```diff
@@ -134,6 +134,16 @@ func (n *Node) Copy() *Node {
 	return nnode
 }
 
+// UpdateNodeLink return a copy of the node with the link name set to point to
+// that. If a link of the same name existed, it is removed.
+func (n *Node) UpdateNodeLink(name string, that *Node) (*Node, error) {
+	newnode := n.Copy()
+	err := newnode.RemoveNodeLink(name)
+	err = nil // ignore error
+	err = newnode.AddNodeLink(name, that)
+	return newnode, err
```

Comment:
That's why the Node is not mutated by the function. It is copied on a new instance which is modified. In fact, that was already done like that in another part of the code (`update()` function in ` fuse/ipns/ipns_unix.go`).